forch/forchestrator.py

```diff
@@ -436,9 +445,43 @@ def _restore_states(self):
     def _restore_faucet_config(self, timestamp, config_hash):
         config_info, faucet_dps, _ = self._get_faucet_config()
         self._update_config_warning_varz()
-        assert config_hash == config_info['hashes'], 'config hash info does not match'
+
+        if config_hash == config_info['hashes']:
+            self._attempt_cancel_config_hash_clash_timer()
+        else:
+            LOGGER.warning('Config hash does not match')
+            self._attempt_start_config_hash_clash_timer()
+
         self._faucet_collector.process_dataplane_config_change(timestamp, faucet_dps)
 
+    def _attempt_start_config_hash_clash_timer(self):
+        if self._config_hash_clash_timer and self._config_hash_clash_timer.is_alive():
+            return
+        self._config_hash_clash_timer = threading.Timer(
```

Comment:
you should reuse one of the _existing_ heartbeat schedulers for this -- creating lots of little threads ends up being very messy in the long run.

Reply:
Using _faucet_state_scheduler now..